packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -169,42 +176,49 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
   return getGradleVersionFor(androidPluginVersion ?? 'unknown');
 }
 
+/// Returns the gradle file from the top level directory.
+/// The returned file is not guaranteed to be present.
+File getGradleWrapperFile(Directory directory) {
+  return directory.childDirectory(gradleDirectoryName)
+      .childDirectory(gradleWrapperDirectoryName)
+      .childFile(gradleWrapperPropertiesFilename);
+}
+
+/// Parses the gradle wrapper distribution url to return a string containing
+/// the version number.
+///
+/// Expected input is of the form '...gradle-7.4.2-all.zip', and the output
+/// would be of the form '7.4.2'.
+String? parseGradleVersionFromDistributionUrl(String? distributionUrl) {
```

Comment:
Does this file need logger.printWarning like getGradleVersion did to help us understand if we cant parse a version?

Reply:
There is a logger warning that triggers in the case that this returns null where it is used in gradle_utils.dart, so unless you preferred it to be refactored to print from in the method itself we should be safe.